parcels/_typing.py

```diff
@@ -0,0 +1,50 @@
+"""
+Typing support for Parcels.
+
+This module contains type aliases used throughout Parcels as well as functions that are
+used for runtime parameter validation (to ensure users are only using the right params).
+
+"""
+
+import ast
+import datetime
+import os
+from typing import Any, Callable, Literal, get_args
+
+
+class ParcelsAST(ast.AST):
+    ccode: str
+
+
+# InterpMethod = InterpMethodOption | dict[str, InterpMethodOption] # (can also be a dict, search for `if type(interp_method) is dict`)
+# InterpMethodOption = Literal[
+#     "nearest",
+#     "freeslip",
+#     "partialslip",
+#     "bgrid_velocity",
+#     "bgrid_w_velocity",
+#     "cgrid_velocity",
+#     "linear_invdist_land_tracer",
+#     "nearest",
+#     "cgrid_tracer",
+# ] # mostly corresponds with `interp_method` # TODO: This should be narrowed. Unlikely applies to every context
```

Comment:
@erikvansebille do you have insight into the contexts in which different interpolation methods are used? Is there additional nuance here?

Reply:
What do you mean with additional nuance? See https://docs.oceanparcels.org/en/latest/examples/tutorial_interpolation.html for an overview of (almost?) all of these options, and how they differ.

Reply:
activate code and add type annotations